### PR TITLE
Add example using Slurm static compute nodes

### PR DESCRIPTION
**Pull request by nick-stroud**
Adds example and test that captures best practices found in the new static nodes cloud docs.

* [About [Slurm] node types](https://cloud.google.com/hpc-toolkit/docs/slurm/node-types)
* [Best practices for static compute nodes](http://cloud/hpc-toolkit/docs/slurm/static-nodes-best-practices)
* [Reconfigure a running cluster](http://cloud/hpc-toolkit/docs/slurm/reconfigure-cluster)
* [Manage static compute nodes](http://cloud/hpc-toolkit/docs/slurm/manage-static-nodes)

Integration test requires creating reservation prior to deployment.

### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

**Review by tpdownes — CHANGES_REQUESTED** [review]
The overall concept and blueprint looks good. There are 2 issues to think over and address:

1. The `make tests` failure for an undefined reservation name. You can look at the A3 blueprint for solutions, though I think we need to do better than that as a follow up task
2. Ensuring the reservation is deleted even when ghpc destroy fails. I suggest one approach, but there is probably some thought/discussion to be had.

File: examples/hpc-slurm-static-v6.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+vars:
+  project_id: ## Set GCP Project ID Here ##
+  deployment_name: static-1
+  region: us-central1
+  zone: us-central1-a
+
+  # For instructions on creating a reservation see:
+  # http://cloud/hpc-toolkit/docs/slurm/static-nodes-best-practices
+  static_reservation_name:  ## Set your reservation name here ##
```
[the remark this answers is not in the document]
Comment (nick-stroud) [reply examples/hpc-slurm-static-v6.yaml]: I have handled in a similar way. Agreed, area for further improvement in the future.

File: community/modules/scheduler/schedmd-slurm-gcp-v6-login/metadata.yaml [hunk 2]
```diff
@@ -18,3 +18,4 @@ spec:
     services: []
 ghpc:
   has_to_be_used: true
+  # inject_module_id: name_prefix
```
[the remark this answers is not in the document]
Comment (nick-stroud) [reply community/modules/scheduler/schedmd-slurm-gcp-v6-login/metadata.yaml]: removed.

File: tools/cloud-build/daily-tests/ansible_playbooks/tasks/rescue_ghpc_failure.yml [hunk 3]
```diff
@@ -39,3 +39,9 @@
     chdir: "{{ workspace }}"
   environment:
     TF_IN_AUTOMATION: "TRUE"
+
+- name: Run post destroy tasks
```
[the remark this answers is not in the document]
Comment (nick-stroud) [reply tools/cloud-build/daily-tests/ansible_playbooks/tasks/rescue_ghpc_failure.yml]: Next I would like to add compact placement to the reservation (future PR), and compact placement does require that VMs are deleted before placement policy is deleted. I have added a block with always to make sure it aways runs. 

I have tested the nested always logic with the following test ansible files.

```
---
- name: Testing nested always
  hosts: 127.0.0.1
  connection: local
  tasks:
  - name: Mock Test Block
    block:
    - name: Mock Test
      ansible.builtin.debug:
        msg: Mock Test
    - name: Trigger failure
      ansible.builtin.fail:
        msg: Test Fails
    always:
    - name: Import Cleanup
      ansible.builtin.include_tasks:
        file: block_testing_task.yml
        apply:
          delegate_to: localhost
```

```
# block_testing_task.yml
---
- name: Cleanup Block
  block:
  - name: Mock Destroy
    ansible.builtin.debug:
      msg: Mock Destroy
  - name: Destroy Fails
    ansible.builtin.fail:
      msg: Destroy Fails
  always:
  - name: Mock Post Destroy
    ansible.builtin.debug:
      msg: Mock Post Destroy
```

File: tools/cloud-build/daily-tests/ansible_playbooks/post-destroy-tasks/delete-reservation.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  ansible.builtin.assert:
+    that:
+    - project is defined
+    - zone is defined
+    - custom_vars.reservation_name is defined
+
+- name: Create Reservation
+  register: create_reservation_result
+  changed_when: create_reservation_result.rc == 0
+  ansible.builtin.command: gcloud compute reservations delete {{ custom_vars.reservation_name }} --project={{ project }}  --zone={{ zone }} --quiet
```
[the remark this answers is not in the document]
Comment (nick-stroud) [reply tools/cloud-build/daily-tests/ansible_playbooks/post-destroy-tasks/delete-reservation.yml:25]: Ack.

**Review by tpdownes — APPROVED** [review]

**Pull request merged**